pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -71,98 +62,58 @@
    * @param pinotHelixResourceManager The resource checker used to interact with Helix
    * @param config The controller configuration object
    */
-  public SegmentStatusChecker(PinotHelixResourceManager pinotHelixResourceManager, ControllerConf config, ControllerMetrics metricsRegistry) {
-    _pinotHelixResourceManager = pinotHelixResourceManager;
+  public SegmentStatusChecker(PinotHelixResourceManager pinotHelixResourceManager, ControllerConf config,
+      ControllerMetrics metricsRegistry) {
+    super("SegmentStatusChecker", config.getStatusCheckerFrequencyInSeconds(), pinotHelixResourceManager);
     _helixAdmin = pinotHelixResourceManager.getHelixAdmin();
     _config = config;
-    _segmentStatusIntervalSeconds = config.getStatusCheckerFrequencyInSeconds();
     _waitForPushTimeSeconds = config.getStatusCheckerWaitForPushTimeInSeconds();
     _metricsRegistry = metricsRegistry;
     HttpConnectionManager httpConnectionManager = new MultiThreadedHttpConnectionManager();
-
-    _executorService = Executors.newSingleThreadScheduledExecutor(new ThreadFactory() {
-      @Override
-      public Thread newThread(Runnable runnable) {
-        Thread thread = new Thread(runnable);
-        thread.setName("Segment status checker");
-        return thread;
-      }
-    });
   }
 
-  /**
-   * Starts the segment status checker.
-   */
-  public void start() {
-    if (_segmentStatusIntervalSeconds == -1) {
-      LOGGER.warn("Segment status check interval is -1, status checks disabled.");
-      return;
-    }
-
+  @Override
+  public void init() {
     setStatusToDefault();
-
-    _executorService.scheduleWithFixedDelay(new Runnable() {
-      @Override
-      public void run() {
-        try {
-          updateSegmentMetrics();
-        } catch (Throwable e) {
-          // catch all errors to prevent subsequent exeuctions from being silently suppressed
-          // Ref: https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay-java.lang.Runnable-long-long-java.util.concurrent.TimeUnit-
-          LOGGER.warn("Caught exception while running segment status checker", e);
-        }
-      }
-    }, SegmentCheckerDefaultIntervalSeconds, _segmentStatusIntervalSeconds, TimeUnit.SECONDS);
   }
 
+  @Override
+  public void onBecomeNotLeader() {
+    super.onBecomeNotLeader();
```

Comment:
why invoke super? Is every sub-class required to call super? why?
In the base class, do whatever is needed on gaining leadership, and then call the subclass method.